syntax = "proto3";

package roboapp;

import "buf/validate/validate.proto";

message Target {
  int32 x = 1 [(buf.validate.field).int32 = {
    gte: 0
    lte: 1280
  }];
  int32 y = 2 [(buf.validate.field).int32 = {
    gte: 0
    lte: 720
  }];
  int32 distance = 3 [(buf.validate.field).int32.gte = 0];
}

message DamagePanelMessage {
  optional Target target = 1;
}
